hw/ip/adc_ctrl/data/adc_ctrl_testplan.hjson

```diff
@@ -12,22 +12,279 @@
+
+          - From monitored ADC data predict when filters should be triggered.
+          - Confirm this by reading the filter status register.
+          '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: filters_polled_fixed
+      desc: '''
+          Verify filter functionality by polling filter status register (fixed filter parameters).
+
+          **Stimulus**:
+
+          For a number of iterations:
+          - Reset and configure DUT with fixed filter parameters, filters enabled
+          but no interrupts or wake events.
+          - Generate ADC data stream.
+
+          **Checks**:
+
+          - From monitored ADC data predict when filters should be triggered.
+          - Confirm this by reading the filter status register.
+          '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: filters_interrupt
+      desc: '''
+          Verify filter interrupt functionality.
+
+          **Stimulus**:
+
+          For a number of iterations:
+          - Reset and configure DUT with randomized filter parameters, filters and interrupts enabled.
+          - Generate ADC data stream.
+
+
+          **Checks**:
+
+          - From monitored ADC data predict when filters should be triggered.
+          - Confirm this by reading the filter status register and interrupt status register.
+          - Confirm correct interrupt sample value has been captured in ADC sample registers(s)
+          - Check interrupt signal operates as expected.
+          '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: filters_interrupt_fixed
+      desc: '''
+          Verify filter interrupt functionality (fixed filter parameters).
+
+          **Stimulus**:
+
+          For a number of iterations:
+          - Reset and configure DUT with fixed filter parameters, filters and interrupts enabled.
+          - Generate ADC data stream.
+
+
+          **Checks**:
+
+          - From monitored ADC data predict when filters should be triggered.
+          - Confirm this by reading the filter status register and interrupt status register.
+          - Confirm correct interrupt sample value has been captured in ADC sample registers(s)
+          - Check interrupt signal operates as expected.
+          '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: filters_wakeup
+      desc: '''
+          Verify filter wakeup functionality
+
+          **Stimulus**:
+          For a number of iterations:
+          - Reset and configure DUT with randomized filter parameters,
+          filters and wakeup enabled, low power sampling mode.
+          - Generate ADC data stream.
+
+          **Checks**:
+
+          - From monitored ADC data predict when filters should be triggered.
+          - Confirm this by reading the filter status register.
+          - Check wakeup signal operates as expected.
+          '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: filters_wakeup_fixed
```

Comment:
@sriyerg 

These several filter tests are very similar. After the first one, should we only capture the differences in the other entries to reduce the duplication? WDYT?

Reply:
Agreed, but fine either way. Testpoints to refer to other testpoints for description if similar stimulus / checks apply.

Reply:
Thanks Sri